packages/cdktf/lib/terraform-asset.ts

```diff
@@ -92,14 +99,20 @@ export class TerraformAsset extends Resource {
   protected onSynthesize(session: ISynthesisSession) {
     const manifest = session.manifest as Manifest;
     const stackManifest = manifest.forStack(this.stack);
-
-    const targetPath = path.join(
+    const basePath = path.join(
       session.outdir,
       stackManifest.synthesizedStackPath,
-      "..",
-      this.path
+      ".."
     );
 
+    // Cleanup existing assets
+    const previousVersionsFolder = path.join(basePath, this.namedFolder);
+    if (fs.existsSync(previousVersionsFolder)) {
+      fs.rmdirSync(previousVersionsFolder, { recursive: true });
+    }
```

Comment:
That's probably an issue on windows, since `namedFolder` is using a [posix path](https://nodejs.org/api/path.html#path_windows_vs_posix)

Reply:
Most of the time Windows understands posix paths.

I'm running on Windows, so just let me know if you need help testing something. (It's much easier testing a pre-release than building from source).

Reply:
Ok - then let's go with the existing approach